sample-app/src/main/java/at/bitfire/cert4android/demo/MainActivity.kt

```diff
@@ -116,58 +115,95 @@ class MainActivity : ComponentActivity() {
                                 model.resultMessage.value = null
                             }
                     }
-                    SnackbarHost(snackBarHostState)
                 }
+
+                if (pendingDecision != null)
+                    TrustCertificateDialog(
+                        certificateDetails = pendingDecision.certificateDetails,
+                        onSetTrustDecision = { actualDecision ->
+                            model.registerUserDecision(pendingDecision, actualDecision)
+                        }
+                    )
+
+                SnackbarHost(
+                    snackBarHostState,
+                    modifier = Modifier.align(Alignment.BottomCenter)
+                )
             }
         }
     }
 
 
-    class Model(application: Application): AndroidViewModel(application) {
+    class Model(application: Application) : AndroidViewModel(application) {
 
-        val appInForeground = MutableStateFlow(true)
         val resultMessage = MutableLiveData<String>()
 
+        class PendingDecision(
+            val certificateDetails: CertificateDetails,
+            val userDecision: CompletableDeferred<Boolean>
+        )
+        private val _pendingDecisionFlow = MutableStateFlow<PendingDecision?>(null)
+        val pendingDecisionFlow: StateFlow<PendingDecision?> = _pendingDecisionFlow
```

Comment:
Got rid of the `@Volatile userDecision` 🙂 

Seems as if it works but please double-check

Reply:
We added it premptively. See https://github.com/bitfireAT/cert4android/pull/50#discussion_r1963309964

Reply:
I mean got rid of the `userDecision` at all

Reply:
I think PendingDecision can be a data class no?

Reply:
I had that thought, too. However the question is what for? We don't use comparison, so we don't need `equals()`, and we don't want to print the values, so no `toString()`. Is there anything else how we [benefit from a data class](https://kotlinlang.org/docs/data-classes.html)?

Reply:
mmh, true. I got so used to the notion of small simple classes without methods automatically being data classes...